app/src/main/java/uw/hcrlab/kubi/lesson/prompts/SelectPrompt.java

```diff
@@ -70,72 +65,96 @@ public void onStart() {
         super.onStart();
 
         // Load the audio resources
-        Activity activity = getActivity();
-
-        mPronunciations = new HashMap<>();
-
         for(PromptData.Option option: this.data.options) {
-            String url = App.getAudioURL(option.title);
-
-            if(url != null) {
-                String audioUrl = mProxy.getProxyUrl(url);
-                mPronunciations.put(createOptionTag(option.idx), MediaPlayer.create(activity, Uri.parse(audioUrl)));
-            }
+            robot.loadPronunciation(option.title);
         }
+    }
 
+    @Override
+    public void onResume() {
+        super.onResume();
+
+        final String[] parts = this.data.PromptText.split("[“”]");
+
+        if(parts.length > 1) {
```

Comment:
This takes a NAME prompt like 

> Translate "a man"

and returns

> a man

so that it can be displayed in the thought bubble. This way the robot says what to translate and shows a text version of what to translate.